### PR TITLE
add has_property to Fake

### PR DESCRIPTION
**Pull request by jsatt**
Useful for raising on property access.

### CODE REVIEW

File: fudge/__init__.py [hunk 1]
```diff
@@ -888,6 +892,25 @@ def has_attr(self, **attributes):
         self._attributes.update(attributes)
         return self
 
+    def has_property(self, **properties):
```
Comment (kumar303) [inline fudge/__init__.py:895]: `has_properties()` would read a little better to me since you can pass in multiple

**Comment by kumar303** [comment]
How is this different than `has_attr()` ?

**Comment by kumar303** [comment]
What I mean is, can you not just fix `has_attr()` so it handles property-style access?

**Comment by jsatt** [comment]
Since the properties are handled different than the attrs in `__getattribute__` you need some indicator that the function you provided for the property needs to be called rather than simply returned, hence (_properties vs _attributes). Providing a separate setter for this creates a clear interface to differentiate the two patterns, but if we can come up with a way to provide this clarity with `has_attr`, I'm open.

**Comment by kumar303** [comment]
ok, I see. That makes sense now. Looks good.

**Pull request merged**